plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php

```diff
@@ -223,10 +223,17 @@ protected function render( $attributes, $content, $block ) {
 			return '';
 		}
 
-		$product_name = $product->get_name();
-		$product_html = $is_stepper_style ? $this->add_steppers( $product_html, $product_name ) : $product_html;
+		// The quantity input might be hidden if the min and max quantities
+		// returned by `woocommerce_quantity_input_args` are the same.
+		$has_visible_input = Utils::has_visible_quantity_input( $product_html );
+		if ( $has_visible_input ) {
+			$product_name = $product->get_name();
+			$product_html = $is_stepper_style ? $this->add_steppers( $product_html, $product_name ) : $product_html;
+			$product_html = $is_stepper_style ? $this->add_stepper_classes_to_add_to_cart_form_input( $product_html ) : $product_html;
```

Comment:
This is nitpick but I found this section a bit _busy_. Seems like it could be presented like this, wdyt:

```suggestion
		if ( $has_visible_input && $is_stepper_style) {
			$product_name = $product->get_name();
			$product_html = $this->add_steppers( $product_html, $product_name );
			$product_html = $this->add_stepper_classes_to_add_to_cart_form_input( $product_html );
```

Reply:
Oh, much better! :100:

Reply:
Actually, I went a step further and simplified the code a bit more: 119ab5144374bbc42dda0e8210a2931e15577748.